dockerfiles/artifactory/Dockerfile

```diff
@@ -1,19 +1,33 @@
-FROM node:12-buster-slim
+FROM snyk/ubuntu as base
 
 MAINTAINER Snyk Ltd
 
+USER root 
+
+RUN apt update && apt install -y make python gcc
+
+USER node 
+
+ENV NPM_CONFIG_PREFIX=/home/node/.npm-global
+
+ENV PATH=$PATH:/home/node/.npm-global/bin 
+
 RUN npm install --global snyk-broker
 
+USER root 
+
 RUN apt-get update && apt-get install -y ca-certificates
 
-# Don't run as root
-WORKDIR /home/node
-USER node
+FROM snyk/ubuntu
 
-# Generate default accept filter
-RUN broker init artifactory
+ENV PATH=$PATH:/home/node/.npm-global/bin 
+
+COPY --from=base /home/node/.npm-global /home/node/.npm-global
 
+USER root
```

Comment:
Previously we explicitly called out `# Don't run as root`, do we want to do the same here?

Reply:
So fixed it. You'll still see the script running as `root` in the first stage, but don't worry about it.
These dockerfiles use what's called a [multistage build in docker](https://docs.docker.com/develop/develop-images/multistage-build/). The first phase, which installs everything has to be when running as `root`, but this is just the build stage.
A few lines later, we start from scratch (from a clean `snyk/ubuntu` image to be exact), and just copy over everything we installed before. In this (final) stage, the user is no longer `root`, but `node`, which is the only thing that matters for the final container that is deployed.